website/docs/concepts/provider_lifecycles.mdx

```diff
@@ -0,0 +1,87 @@
+---
+title: Provider Lifecycles
+---
+
+## When does my Provider get created and disposed?
+
+The states that all different types of providers can go through are the same:
+
+- Uninitialized
+- Creating
+- Alive
+- Disposing
+
+### Uninitialized
+
+An uninitialized provider does not take up any memory since its state is not initialized.
+It will stay that way until an Alive provider or the UI reads or watches it.
+
+### Creating -> Alive
+
+When an uninitialized provider is being read or watched it will be created.
+
+When a provider is created your provider creation function will be run. Any
+providers that you read or watch will be created as needed and their state will be retrieved.
+
+If there are any circular dependencies during this creation process Riverpod will throw an error.
+The best way to fix this error is to redesign your dependencies to have a uni-directional dataflow.
```

Comment:
I don't think that's the correct place to discuss about this. Maybe within the "combining providers" part.

Uni-directional dataflow is not related to life-cycles